pkgs/development/libraries/zlib/default.nix

```diff
@@ -104,7 +104,7 @@ stdenv.mkDerivation (finalAttrs: {
     # Non-typical naming confuses libtool which then refuses to use zlib's DLL
     # in some cases, e.g. when compiling libpng.
     + lib.optionalString (stdenv.hostPlatform.isMinGW && shared) ''
-      ln -s zlib1.dll $out/bin/libz.dll
+      ln -s $out/lib/libz.dll $out/lib/zlib1.dll
```

Comment:
Do we really want to lose the link in `$out/bin`? That matches the default search path for Windows shared libs.